src/Sulu/Bundle/AdminBundle/Resources/js/containers/Application/Application.js

```diff
@@ -15,15 +15,18 @@ type Props = {
 @observer
 export default class Application extends React.PureComponent<Props> {
     render() {
+        const {router} = this.props;
+
         return (
             <div>
                 <Toolbar />
                 <main className={applicationStyles.main}>
-                    {this.props.router.currentRoute &&
+                    {router.currentRoute &&
                         <ViewRenderer
                             key={this.props.router.currentRoute.name}
                             name={this.props.router.currentRoute.view}
-                            parameters={this.props.router.currentParameters} />
+                            parameters={this.props.router.currentParameters}
+                            router={router} />
```

Comment:
This made me ask the question if we should omit the `parameters` prop above as well. Why should we pass it, if the `View` could get all information it needs from the `Router` directly? That would also mean that the `View`s have a very clear interface - only the router is passed.

Reply:
On the other handside it is nicer to write `Views` with proper props being passed for a developer. But then it's the question how to pass the query parameters? Merging the query parameters might be convenient in some cases, but might allow to override parameters via query parameters even if they are not supposed to be used like that.

Reply:
I've change that implementation for now in the other PR, which should be merge before this one: https://github.com/sulu/sulu/pull/3507/commits/b5574dd523ee0413a16a2dbd453439f3527fd0a1

We can discuss that, and revert the changes if we decide to handle this in a different way.